sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -320,7 +322,26 @@ def restrict(self, support, indices=False):
             Boolean mask or list of indices (as returned by the get_support
             member of feature selectors).
         indices : boolean, optional
-            Whether support is a list of indices.
+            Whether support is a list of indices. 
+
+        Returns
+        -------
+        self
+
+        Examples
+        --------
+        >>> from sklearn.feature_extraction import DictVectorizer
+        >>> from sklearn.feature_selection import SelectKBest, chi2
+        >>> v = DictVectorizer()
+        >>> D = [{'foo': 1, 'bar': 2}, {'foo': 3, 'baz': 1}]
+        >>> X = v.fit_transform(D)
+        >>> support = SelectKBest(chi2, k=2).fit(X, [0, 1])
+        >>> v.get_feature_names()
+        ['bar', 'baz', 'foo']
+        >>> v.restrict(support.get_support()) # doctest: +ELLIPSIS
+        DictVectorizer(dtype=<...
```

Comment:
please keep as much of the text as possible. The current one looks odd. I'm surprised you say there is a ``sort`` argument on python 3.4 but not on 2.7..... That can't be right.